nixos/modules/services/networking/dsnet.nix

```diff
@@ -0,0 +1,364 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.dsnet;
+  settingsFormat = pkgs.formats.json { };
+  patchFile = settingsFormat.generate "dsnet-patch.json" cfg.settings;
+in
+{
+  options.services.dsnet = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = "Enable dsnet as a systemd service.";
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = (pkgs.callPackage ./dsnet.nix { });
+      description = "The dsnet package to use.";
+    };
+
+    settings = lib.mkOption {
```

Comment:
I feel like we might need to reassess how many options we model here. RFC 42 says that you should only model the options that are either necessary or most often used, as more options equate more maintenance costs in the future.

Reply:
Is there a way I can allow options through without explicitly defining them? I take it yes, that's what freeformType (or something) does.

If so I will remove the options that I don't think will be changed or used much to reduce maintenance